src/main/java/com/simibubi/create/foundation/events/ClientEvents.java

```diff
@@ -322,17 +322,14 @@ public static void getFogDensity(ViewportEvent.RenderFog event) {
 
 		ItemStack divingHelmet = DivingHelmetItem.getWornItem(entity);
 		if (!divingHelmet.isEmpty()) {
-			if (FluidHelper.isWater(fluid)) {
+			if (FluidHelper.isWater(fluid) && AllConfigs.server().equipment.shouldScaleWaterFogDistance.get()) {
 				event.scaleFarPlaneDistance(6.25f);
-				event.setCanceled(true);
-				return;
 			} else if (FluidHelper.isLava(fluid) && NetheriteDivingHandler.isNetheriteDivingHelmet(divingHelmet)) {
 				event.setNearPlaneDistance(-4.0f);
 				event.setFarPlaneDistance(20.0f);
-				event.setCanceled(true);
-				return;
 			}
 		}
+		event.setCanceled(true);
```

Comment:
The event.setCanceled call shouldn’t be moved out of the ifs, as that would cancel the event even if Create doesn’t do anything, which *will* break other mods. (Including Alex’s Caves’ own fog rendering for its fluids)